test_conformance/subgroups/test_workgroup_non_uniform_vote.cpp

```diff
@@ -407,48 +256,60 @@ struct run_for_type
+            test<T, VOTE<T, NonUniformVoteOp::elect, work_items_mask>,
+                 GWS_NON_UNIFORM, LWS_NON_UNIFORM>::run(device_, context_,
+                                                        queue_, num_elements_,
+                                                        "test_elect",
+                                                        elect_source, 0,
+                                                        useCoreSubgroups_,
+                                                        required_extensions_,
+                                                        work_items_mask);
 
         return error;
     }
 
-    template <typename T> int run_nu_any()
+    template <typename T, unsigned int work_items_mask> int run_nu_any()
     {
         int error =
-            test<T, AAN<T, NonUniformVoteOp::any>, GWS_NON_UNIFORM,
-                 LWS_NON_UNIFORM>::run(device_, context_, queue_, num_elements_,
-                                       "test_non_uniform_any",
-                                       non_uniform_any_source, 0,
-                                       useCoreSubgroups_, required_extensions_);
+            test<T, VOTE<T, NonUniformVoteOp::any, work_items_mask>,
+                 GWS_NON_UNIFORM, LWS_NON_UNIFORM>::run(device_, context_,
+                                                        queue_, num_elements_,
+                                                        "test_non_uniform_any",
+                                                        non_uniform_any_source,
+                                                        0, useCoreSubgroups_,
+                                                        required_extensions_,
+                                                        work_items_mask);
         return error;
     }
 
-    template <typename T> int run_nu_all()
+    template <typename T, unsigned int work_items_mask> int run_nu_all()
     {
         int error =
-            test<T, AAN<T, NonUniformVoteOp::all>, GWS_NON_UNIFORM,
-                 LWS_NON_UNIFORM>::run(device_, context_, queue_, num_elements_,
-                                       "test_non_uniform_all",
-                                       non_uniform_all_source, 0,
-                                       useCoreSubgroups_, required_extensions_);
+            test<T, VOTE<T, NonUniformVoteOp::all, work_items_mask>,
+                 GWS_NON_UNIFORM, LWS_NON_UNIFORM>::run(device_, context_,
+                                                        queue_, num_elements_,
+                                                        "test_non_uniform_all",
+                                                        non_uniform_all_source,
+                                                        0, useCoreSubgroups_,
+                                                        required_extensions_,
+                                                        static_cast<int>(
```

Comment:
BTW, these `run_*` functions look very similar. I see the mask is now a function parameter If you want to reduce code duplication further and avoid such copy/paste issues, you could create a base function, possibly with the following signature or something close to it:

```c++
template <typename T, unsigned int work_items_mask, NonUniformVoteOp op>
int run_impl(char const* kernel_name, char const* source)
```

It might also be possible to make `work_items_mask` a non-template argument by slightly modifying `gen`/`chk` functions and making `test::run` have extra variadic template arguments that are passed to `gen`/`chk`. This would reduce the number of template instantiation in the code. Just a thought.

Reply:
fixed